packages/apputils/src/sessioncontext.tsx

```diff
@@ -1297,13 +1348,26 @@ export const sessionContextDialogs: ISessionContext.IDialogs = {
       Dialog.okButton({ label: trans.__('Select') })
     ];
 
+    const selectPreferredKernel = this._settings?.get('selectPreferredKernel')
+      .composite as boolean;
+
     const dialog = new Dialog({
       title: trans.__('Select Kernel'),
       body: new Private.KernelSelector(sessionContext, translator),
-      buttons
+      buttons,
+      checkbox: {
+        label: trans.__('Always select the preferred kernel'),
+        caption: trans.__(
+          'Remember my choice and always select the preferred kernel'
+        ),
+        checked: selectPreferredKernel
+      }
```

Comment:
We should not show the checkbox if there is no settings plugin.

```suggestion
      checkbox: this._settings && {
        label: trans.__('Always select the preferred kernel'),
        caption: trans.__(
          'Remember my choice and always select the preferred kernel'
        ),
        checked: selectPreferredKernel
      }
```

Reply:
Actually, Setting Plugin is required for ISessionDialogs plugin

```
/**
 * The default session context dialogs extension.
 */
const sessionDialogs: JupyterFrontEndPlugin<ISessionContextDialogs> = {
  id: '@jupyterlab/apputils-extension:sessionDialogs',
  provides: ISessionContextDialogs,
  requires: [ISettingRegistry, ITranslator],
  autoStart: true,
  activate: async (
    app: JupyterFrontEnd,
    settingRegistry: ISettingRegistry,
    translator: ITranslator
  ) => {
    const settings = await settingRegistry.load(
      '@jupyterlab/notebook-extension:tracker'
    );
    return new SessionContextDialogs(settings, translator);
  }
};
```

I think it is better to make it optional, because just it is possible.

Reply:
However I have found that `@jupyterlab/apputils-extension:settings` is required by so many other plugins. So making is optional for '@jupyterlab/apputils-extension:sessionDialogs' is not meaningful.

Reply:
It is still possible to make it optional, but it cannot be tested by hand (or by some kind of integration test)

Reply:
This is what i got when disabling settings plugin
![image](https://user-images.githubusercontent.com/4434752/218652952-0b39d895-32b2-4831-b3a7-8adb04798f79.png)